controllers/etcdcluster_controller.go

```diff
@@ -274,6 +494,27 @@ func initialClusterMembers(cluster *etcdv1alpha1.EtcdCluster) []etcdv1alpha1.Ini
 	return members
 }
 
+func initialClusterMembersFromMemberList(cluster *etcdv1alpha1.EtcdCluster, members []etcdclient.Member) []etcdv1alpha1.InitialClusterMember {
+	initialMembers := make([]etcdv1alpha1.InitialClusterMember, len(members))
+	for i, member := range members {
+		name, err := peerNameForMember(member)
+		if err != nil {
+			// We were not able to figure out what this peer's name should be. There is precious little we can do here
+			// so skip over this member.
+			continue
+		}
+		initialMembers[i] = etcdv1alpha1.InitialClusterMember{
+			Name: name,
+			// The member actually will always have at least one peer URL set (in the case where we're inspecting a
+			// member in the membership list that has never joined the cluster that is the only thing we'll see. But
+			// using that is actually painful as it's often strangely escaped. We bypass that issue by regenerating our
+			// expected url from the name.
```

Comment:
I tried it and my URL ended up with `%F` in it a lot. I figured i could try and un-escape it but I wasn't certain if it would do that every time so I just generated the URL again.